internal/metamorph/store/postgresql/postgres.go

```diff
@@ -377,18 +389,36 @@ func (p *PostgreSQL) GetSeenOnNetwork(ctx context.Context, since time.Time, limi
 		,raw_tx
 		,locked_by
      	,merkle_path
-		,retries
-		FROM metamorph.transactions
-		WHERE locked_by = $5
-		AND status = $1
-		AND inserted_at_num > $2
-		ORDER BY inserted_at_num DESC
-		LIMIT $3 OFFSET $4;`
+		,retries;`
+
+	tx, err := p.db.Begin()
+	if err != nil {
+		return nil, err
+	}
 
-	rows, err := p.db.QueryContext(ctx, q, metamorph_api.Status_SEEN_ON_NETWORK, since.Format(numericalDateHourLayout), limit, offset, p.hostname)
+	_, err = tx.Exec(`SELECT hash 
```

Comment:
Is this query intended to lock the rows?

I think you could do all in one query like this

```sql
`UPDATE metamorph.transactions
		SET locked_by=$5
		FROM (
			SELECT hash
			FROM metamorph.transactions
			WHERE (locked_by = $5 OR locked_by = 'NONE')
			AND status = $1
			AND inserted_at_num > $2
			ORDER BY inserted_at_num DESC
			LIMIT $3 OFFSET $4
                        FOR UPDATE SKIP LOCKED
                 ) as t
		WHERE metamorph.transactions.hash = t.hash
		RETURNING
	     stored_at
		,announced_at
		,mined_at
		,t.hash
		,status
		,block_height
		,block_hash
	@@ -377,18 +389,36 @@ func (p *PostgreSQL) GetSeenOnNetwork(ctx context.Context, since time.Time, limi
		,raw_tx
		,locked_by
     	,merkle_path
		,retries;`
```

As it is done here: https://github.com/bitcoin-sv/arc/blob/eb27e0619cb2955589bdfead40ec838ab3c4c89d/internal/metamorph/store/postgresql/postgres.go#L317

Reply:
I'm not sure. That's how we did earlier and it caused deadlock right? 

What if we are updating the same rows? I guess that would be less likely 🤔

Reply:
Ok, let's try it like this